staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_width.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/klog/v2"
+	"math"
+	"net/http"
+	"net/url"
+)
+
+func newListWidthEstimator(countFn objectCountGetterFunc) WidthEstimatorFunc {
+	estimator := &listWidthEstimator{
+		countFn: countFn,
+	}
+	return estimator.estimate
+}
+
+type listWidthEstimator struct {
+	countFn objectCountGetterFunc
+}
+
+func (e *listWidthEstimator) estimate(r *http.Request) Width {
+	requestInfo, ok := apirequest.RequestInfoFrom(r.Context())
+	if !ok {
+		// no RequestInfo should never happen, but to be on the safe side
+		// let's return minimumSeatsList
+		return Width{Seats: maximumSeats}
```

Comment:
Comment says minimum, code says maximum.
Which is "safe"?  Using the maximum is safe as far as protecting against server overload goes.
No misjudgement is "safe" as far as delivering fairness goes.

Reply:
This should  never happen in produciton TBH. We should log an error if that happens (or expose a metric or sth).

Maxium is safe for protection of overload (the comment should be update).
I agree misjudgement is not safe for fairness, but as this should never really happen, we can tolerate this.

Reply:
OK